### DISCUSSION TITLE
Should components be able to nest other components?

### DISCUSSION THREAD

**Discussion by adamghill** [django-commons/django-unicorn#90] [Ideas]
How would this work? Does it seem like it would be useful in the future?

**Comment by brmc** [comment]
Hi, I saw your post on reddit and it reminded me of a proof-of-concept i did a few months ago: https://github.com/brmc/django-sfc

I don't plan on doing anything else with the project, but you might get some ideas, specifically regarding this discussion on nesting components

My primary motivation sounds similar to yours in some respects:

* Create reusable interactive components that can be composed of other components 
* Avoid JS build chains as much as possible
* Leverage advantages of both server-side rendering and SPAs with minimal effort

I didn't plan to create a full framework like you're doing though. I just wanted to see how far i could go and keep things minimalist

There's no readme in my repo, but you can see a reddit post with more details and an explanation of how to use it: https://www.reddit.com/r/django/comments/g0ezsf/singlefile_components_i_was_experimenting_with_an/ 

Feel free to steal, ignore, or discuss. I'm fine with any of the above :)

**Reply by adamghill** [reply]
I really appreciate you posting this, I kept nodding my head while reading your replies, especially

>i was never thinking i could replace any of the major (or, hell, any of the less-than-major) libraries. My thinking is that there is a hole in the "market" for simple interfaces (in other words, targeting the "incremental adoption" use cases that vue and react push) that use some javascript but don't need the added complexity of a different development environment. I mean, How many times a week do people ask how to integrate Vue/Angular/React into django? Shouldn't this be easier, simpler, and more obvious to new comers?
>I also like the idea of working with just html, css, and javascript, without any transpiling

I'm skimming through your code now and will reach out if I have any questions. Thanks!

**Reply by brmc** [reply]
sounds good. i hope you find it helpful

be warned, there are definitely some questionable practices in there since i was just tinkering around XD

i wasn't planning on going any further with it because i didn't want to dedicate the time necessary to bring it to maturity, but if there are some tickets you'd like tackled, let me know, and i'll see what i can do in my spare time

**Comment by adamghill** [comment]
I'm currently working through a prototype in https://github.com/adamghill/django-unicorn/pull/107, but it's not pretty. Some pieces work already (e.g. child component can update a parent component's data), but lots of edge cases to work through.

**Comment by adamghill** [comment]
This has now been released in 0.15.1. Documentation: https://www.django-unicorn.com/docs/child-components/.
